packages/aws-cdk/lib/docker.ts

```diff
@@ -18,38 +18,71 @@ import { PleaseHold } from './util/please-hold';
  *
  * As a workaround, we calculate our own digest over parts of the manifest that
  * are unlikely to change, and tag based on that.
+ *
+ * When running in CI, we pull the latest image first and use it as cache for
+ * the build. Generally pulling will be faster than building, especially for
+ * Dockerfiles with lots of OS/code packages installation or changes only in
+ * the bottom layers. When running locally chances are that we already have
+ * layers cache available.
+ *
+ * CI is detected by the presence of the `CI` environment variable or
+ * the `--ci` command line option.
  */
-export async function prepareContainerAsset(asset: ContainerImageAssetMetadataEntry, toolkitInfo: ToolkitInfo): Promise<CloudFormation.Parameter[]> {
+export async function prepareContainerAsset(asset: ContainerImageAssetMetadataEntry,
+                                            toolkitInfo: ToolkitInfo,
+                                            ci?: boolean): Promise<CloudFormation.Parameter[]> {
   debug(' 👑  Preparing Docker image asset:', asset.path);
 
   const buildHold = new PleaseHold(` ⌛ Building Docker image for ${asset.path}; this may take a while.`);
   try {
+    const ecr = await toolkitInfo.prepareEcrRepository(asset.id);
+    const latest = `${ecr.repositoryUri}:latest`;
+
+    let loggedIn = false;
+
+    // In CI we try to pull latest first
+    if (ci === true || (process.env.CI && ci !== false)) {
```

Comment:
Don't do the CI autodetection here, move it up to `cdk.ts`.

Reply:
Yep. I have a script that recreates the 'cdk --help' part of the tools section whenever we have a new release.